test_bst.py

```diff
@@ -96,3 +96,27 @@ def test_balance_positive():
     test_tree.insert(5)
     test_tree.insert(4)
     assert test_tree.balance() > 0
+
+
+def test_breadth_first_traversal(test_tree_large):
+    expected = [5, 4, 8, 2, 10, 1, 3, 12, 14]
+    for i, val in enumerate(test_tree_large.breadth_first_traversal()):
+        assert val == expected[i]
```

Comment:
Everyone is making this same mistake.  In your case I don't think it's fatal, because I believe your methods to be working, but if your traversal method doesn't actually yield anything, this test will pass, even though it shouldn't.  You need to ensure that values actually comes out of the generator